libvncclient/tls_gnutls.c

```diff
@@ -316,7 +316,7 @@ ReadVeNCryptSecurityType(rfbClient* client, uint32_t *result)
     uint8_t count=0;
     uint8_t loop=0;
     uint8_t flag=0;
-    uint32_t tAuth[256], t;
+    uint32_t tAuth[254], t;
```

Comment:
While I agree that there will very likely never be more than 254 types advertised, setting an arbitrary limit feels a bit wrong. Perhaps we could cast `count` to `size_t` in the comparison instead?

Reply:
It wouldn't make any difference ;-) Per RFB protocol the number of security types is transmitted as a uint8_t so even casting wouldn't change the behaviour. Either we drop the check completely (or check against 0 only) and keep `tAuth` at 256 elements or we choose an arbitrary limit as proposed. Any opinions?

Reply:
It would make the warning go away :)